### PR TITLE
Fix rewrite behavior for abs in sign(x) 

### LINKED ISSUE #27323: rewrite(abs) should call rewrite(Abs)
```py
>>> sign(x).rewrite(abs)
sign(x)
>>> sign(x).rewrite(Abs)
Piecewise((0, Eq(x, 0)), (x/Abs(x), True))
```

This is because `rewrite(x)` just dispatches to the method called `_eval_rewrite_as_x`, and the method here is called `_eval_rewrite_as_Abs` since `Abs` is the name of the class. But most people use `abs`, the built-in function which dispatches to `Abs`. So it should be special-cased in rewrite to call `Abs`.

### PR DESCRIPTION
**Pull request by digkioan**
Fixes #27323.

This PR fixes the behavior of `rewrite` when called with the built-in `abs` function. Previously, `sign(x).rewrite(abs)` would return `sign(x)` without any transformation, while `sign(x).rewrite(Abs)` worked correctly. 

This inconsistency has been resolved by mapping the built-in `abs` function to the `Abs` class within `_eval_rewrite_as_Abs`.

**Key changes:**
- Updated `_eval_rewrite_as_Abs` to handle the built-in `abs` correctly.
- Added tests to verify the behavior of `rewrite` with both `abs` and `Abs`.

* functions
  * Fixed a bug in `rewrite` for `sign(x)` where the built-in `abs` was not mapped to `Abs`.

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -1620,6 +1621,7 @@ cym1 <16437732+cym1@users.noreply.github.com>
 damianos <damianos@semmle.com>
 dandiez <47832466+dandiez@users.noreply.github.com>
 der-blaue-elefant <github@kklein.de>
+digkioan <125358066+digkioan@users.noreply.github.com>
```
Comment (oscarbenjamin) [inline .mailmap:1624]: This line is not needed and should not be here.

Comment (digkioan) [reply .mailmap]: Sorry, I forgot to remove it. Thank you. I will change it immediately.

File: sympy/functions/elementary/miscellaneous.py [hunk 2]
```diff
@@ -642,6 +642,9 @@ def _eval_derivative(self, s):
 
     def _eval_rewrite_as_Abs(self, *args, **kwargs):
         from sympy.functions.elementary.complexes import Abs
+        target = kwargs.get("target", None)
+        if target == abs:
+            target = Abs
```
Comment (oscarbenjamin) [inline sympy/functions/elementary/miscellaneous.py:647]: This doesn't work because this method will not get called for `.rewrite(abs)`.

**Comment by anutosh491** [comment]
Closing as superseded by #27325

**Pull request closed without merging**